src/components/legend.ts

```diff
@@ -144,13 +164,16 @@ export module Components {
 
       let availableWidthForEntries = Math.max(0, (availableWidth - this._padding));
 
-      let entryNames = this._colorScale.domain().slice();
-      entryNames.sort(this.comparator());
-
+      let entryLabels: d3.Map<string> = d3.map<string>();
+      this._colorScale.domain().slice().forEach((d) => {
+        entryLabels.set(this._formatter(d), d);
+      });
+      let formattedEntryNames = entryLabels.keys();
+      let entryNames = formattedEntryNames.sort(this.comparator()).map((d) => entryLabels.get(d));
```

Comment:
Would this work?
`let entryNames = this._colorScale.domain().sort((a, b) => this.comparator()(this.formatter()(a), this.formatter()(b));`

This would get rid of the the `entryLabels` and the `formattedEntryNames` variable and I believe this is still readable